### DISCUSSION TITLE
difference by assigning variables using '==' and 'var: value' in recursion

### DISCUSSION THREAD

**Discussion by wdduncan** [EvgSkv/logica#391] [Q&A]
I am experimenting with using recursion to return RDF triples. In my example use case, I am wanting query for all types of teeth.  

Here are are my basic setup:
```sql
  logica_lib.RunPredicateFromString("""
  @Engine("sqlite");

  Tooth(subject: "tooth", predicate: "type", object: "tooth");
  Tooth(subject: "canine", predicate: "type", object: "tooth");
  Tooth(subject: "upper-left canine", predicate: "type", object: "canine");
  Tooth(subject: "upper-right canine", predicate: "type", object: "canine");
  Tooth(subject: "bottom-left canine", predicate: "type", object: "canine");
  Tooth(subject: "bottom-right canine", predicate: "type", object: "canine");

  ChildOf(x, y) :- *** see #1 and #2 below ***;

  DescendantOf(x, y) :- ChildOf(x, y);
  DescendantOf(x, z) :- ChildOf(x, y), DescendantOf(y, z);
  IsaTooth(x) = y :- y == "tooth", DescendantOf(x, y);

  Teeth(tooth: subject) distinct :- 
    Tooth(subject:, predicate:, object:),
    IsaTooth(subject);
""", 'Teeth')
```

I can define `ChildOf` in two ways.  
1. assign variables using `==`:
```sql
  ChildOf(x, y) :- Tooth(subject:, predicate:, object:), 
    subject == x, 
    predicate == "type", 
    object == y;
```

2. assign value in the predicate:
```sql
  ChildOf(x, y) :- Tooth(subject: x, predicate: "type", object: y);
```

Each returns the same table. 
<img width="181" alt="image" src="https://github.com/user-attachments/assets/faf6e3d4-93f8-489a-acba-53b9f09b4b28">

What is the difference in the two `ChildOf` predicates?

**Comment by EvgSkv** [comment]
@wdduncan As far as I see there is precisely no difference, it compiles to exactly the same internal representation and evenaully same SQL (maybe with different order of constraints in the `WHERE` clause). It is also logically equivalent, of course.

Which way to write is purely a readability question, so chose whatever you find more intuitive.
